modules/tinymce/src/core/main/ts/delete/DeleteElement.ts

```diff
@@ -87,7 +88,12 @@ const isBlock = (editor: Editor, elm: SugarElement<Node>): boolean =>
 const paddEmptyBlock = (elm: SugarElement<Node>): Optional<CaretPosition> => {
   if (Empty.isEmpty(elm)) {
     const br = SugarElement.fromHtml('<br data-mce-bogus="1">');
-    Remove.empty(elm);
+    // Remove all bogus elements except caret
+    Arr.each(Traverse.children(elm), (node) => {
+      if (!isEmptyCaretFormatElement(node)) {
+        Remove.remove(node);
+      }
+    });
```

Comment:
It might be good to introduce a new parameter for `DeleteElement.deleteElement` called `preserve_empty_caret_format` or something, so you can just call `Remove.empty` for all other cases. Seems to unnecessarily slow all cases down when this is only needed for 1.